run_xnat2bids.py

```diff
@@ -61,51 +54,70 @@ def compileArgumentList(arg_dict):
 
+def main():
+    # Instantiate argument parserß
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--config", help="path to user config")
+    args = parser.parse_args()
 
-    return x2b_param_list, slurm_param_list, user
-
-
-# Instantiate argument parserß
-parser = argparse.ArgumentParser()
-parser.add_argument("--config", help="path to user config")
-args = parser.parse_args()
-
-# Load default config file into dictionary
-default_params = load('/gpfs/data/bnc/scripts/x2b_default_config.toml')
+    # Load default config file into dictionary
+    default_params = load('/gpfs/data/bnc/scripts/x2b_default_config.toml')
 
-# Set arglist. If user provides config, merge dictionaries.
-if args.config is None:
-    arglist = default_params
-else:
-    user_params = load(args.config)
-    default_params.update(user_params)
-    arglist = user_params
+    # Set arglist. If user provides config, merge dictionaries.
+    if args.config is None:
+        arglist = default_params
+    else:
+        user_params = load(args.config)
+        default_params.update(user_params)
+        arglist = user_params
 
-# Fetch compiled xnat2bids abd slurm parameter lists
-x2b_param_list, slurm_param_list, user = compileArgumentList(arglist)
-
-# Fetch latest version if not provided
-if(arglist['xnat2bids-args']['version'] == ""):
-    list_of_versions = glob.glob('/gpfs/data/bnc/simgs/brownbnc/*') 
-    latest_version = max(list_of_versions, key=os.path.getctime)
-    version = latest_version.split('-')[-1].replace('.sif', '')
-
-# Define singularity image 
-simg=f"/gpfs/data/bnc/simgs/brownbnc/xnat-tools-{version}.sif"
-
-# Define path to bnc data directory
-data_dir = arglist['xnat2bids-args']['data_dir']
+    # Fetch user credentials 
+    user = input('Enter Username: ')
+    password = getpass('Enter Password: ')
 
-# Run xnat2bids via srun
-subprocess.run(shlex.split(
-    f"srun {' '.join(slurm_param_list)} \
-    singularity exec --contain -B {data_dir} {simg} \
-    xnat2bids {' '.join(x2b_param_list)}"
-))
+    # Fetch compiled xnat2bids abd slurm parameter lists
+    x2b_param_list, slurm_param_list, user = compileArgumentList(arglist, user)
+
+    # Insert username and password into x2b_param_list
+    x2b_param_list.insert(2, f"--user {user}")
+    x2b_param_list.insert(3, f"--pass {password}")
+
+    # Fetch latest version if not provided
+    if(arglist['xnat2bids-args']['version'] == ""):
+        list_of_versions = glob.glob('/gpfs/data/bnc/simgs/brownbnc/*') 
+        latest_version = max(list_of_versions, key=os.path.getctime)
+        version = latest_version.split('-')[-1].replace('.sif', '')
+
+    # Define singularity image 
+    simg=f"/gpfs/data/bnc/simgs/brownbnc/xnat-tools-{version}.sif"
+
+    output = arglist['slurm-args']['output']
+    if not (os.path.exists(os.path.dirname(output))):
+        os.mkdir(os.path.dirname(output))
+
+    # Display formatted command to user
+    new_line = '\n'
```

Comment:
While this maybe a prettier format, I think it may be a good idea to instead have something like this:

```
srun_command = shlex.split(
         f"srun {' '.join(slurm_param_list)} \
         singularity exec --contain -B /gpfs/data/bnc {simg} \
         xnat2bids {' '.join(x2b_param_list)}"
     )

print(f" Running xnat2bids: \n {srun_command})

subprocess.run(srun_command)
```